pkg/java/pom/parse.go

```diff
@@ -111,28 +111,54 @@ func (p *parser) parseRoot(root artifact) ([]types.Library, []types.Dependency,
 
 	var libs []types.Library
 	var deps []types.Dependency
+	processed := map[string]artifact{}
 	uniqArtifacts := map[string]version{}
 
 	// Iterate direct and transitive dependencies
 	for !queue.IsEmpty() {
 		art := queue.dequeue()
 
+		{
+			id := id(art)
+			_, contains := processed[id]
+			if contains {
+				continue
+			} else {
+				processed[id] = art
+			}
+		}
+
 		// Modules should be handled separately so that they can have independent dependencies.
 		// It means multi-module allows for duplicate dependencies.
 		if art.Module {
 			moduleLibs, moduleDeps, err := p.parseRoot(art)
 			if err != nil {
 				return nil, nil, err
 			}
-			libs = append(libs, moduleLibs...)
-			if deps != nil {
+
+			for _, lib := range moduleLibs {
+				_, contains := processed[lib.ID]
+				if contains {
+					continue
+				} else {
+					groupID, artifactID, verion, err := deconstructID(lib.ID)
```

Comment:
ignored map value contains artifact struct which includes artifactID, groupID etc, so it seems `deconstructID()` is not required.